easybuild/tools/easystack.py

```
@@ -0,0 +1,186 @@
+                                raise EasyBuildError(ver_suf_err)
+                            elif 'exclude-labels' in str(yaml_ver_specs) or 'include-labels' in str(yaml_ver_specs):
+                                lab_err = "Easystack specifications of '%s' contain labels. " % str(software)
+                                lab_err += "Labels aren't supported yet."
+                                raise EasyBuildError(lab_err)
+                        except TypeError:
+                            pass
+                        if '*' in str(yaml_version):
+                            raise EasyBuildError(asterisk_err)
+                        # TODO - think of better ID of versions
+                        elif str(yaml_version)[0].isdigit() \
+                                or str(yaml_version)[-1].isdigit():
+                            # creates a sw class instance
+                            try:
+                                yaml_toolchain_name = str(yaml_tc).split('-', 1)[0]
+                                yaml_toolchain_version = str(yaml_tc).split('-', 1)[1]
+                            except IndexError:
+                                yaml_toolchain_name = str(yaml_tc)
+                                yaml_toolchain_version = ''
+
+                            sw = SoftwareSpecs(
+                                software=software, version=yaml_version,
+                                toolchain=yaml_tc, toolchain_name=yaml_toolchain_name,
+                                toolchain_version=yaml_toolchain_version)
+
+                            # append newly created class instance to the list inside EbFromSpecs class
+                            eb.software_list.append(sw)
+                        else:
+                            raise EasyBuildError(wrong_structure_err)
+
+            except (KeyError, TypeError, IndexError):
+                raise EasyBuildError(wrong_structure_err)
```

Don't hide the underlying error please:

```python
            except (KeyError, TypeError, IndexError) as err:
                raise EasyBuildError(wrong_structure_err + ": %s" % err)
```